File: tests/swift_sample_project/SampleProject2/DataHandler.swift
// Sources/SampleProject2/DataHandler.swift

import Foundation

/// Manages loading and saving Item data.
/// Simulates interaction with a data source like a file or database.
public class DataHandler {
    private let dataSourcePath: String

    /// Initializes the DataHandler with the path to the data source.
    public init(dataSourcePath: String) {
        self.dataSourcePath = dataSourcePath
        print("DataHandler initialized for source: \(self.dataSourcePath)")
    }

    /// Simulates loading items from the data source.
    /// - Returns: An array of Item objects.
    public func loadItems() -> [Item] {
        print("Simulating loading items from \(self.dataSourcePath)...")
        let items = [
            Item(itemId: 1, name: "Gadget Alpha", value: 150.75),
            Item(itemId: 2, name: "Widget Beta", value: 85.0),
            Item(itemId: 3, name: "Thingamajig Gamma", value: 210.5),
            Item(itemId: 4, name: "Doohickey Delta", value: 55.2)
        ]
        
        print("Loaded \(items.count) items.")
        return items
    }

    /// Simulates saving processed items back to the data source.
    /// - Parameter items: An array of Item objects to save.
    /// - Returns: True if saving was simulated successfully.
    public func saveItems(items: [Item]) -> Bool {
        print("Simulating saving \(items.count) items to \(self.dataSourcePath)...")
        for item in items {
            print("Saving item: \(item)")
        }
        print("Finished simulating save operation.")
        return true
    }
}

File: tests/csharp_sample_project/DataHandler.cs
// tests/sample_project2/DataHandler.cs

namespace SampleProject2;

/// <summary>
/// Manages loading and saving Item data.
/// Simulates interaction with a data source like a file or database.
/// </summary>
public class DataHandler
{
    private readonly string _dataSourcePath;

    /// <summary>
    /// Initializes the DataHandler with the path to the data source.
    /// </summary>
    public DataHandler(string dataSourcePath)
    {
        _dataSourcePath = dataSourcePath;
        Console.WriteLine($"DataHandler initialized for source: {_dataSourcePath}");
    }

    /// <summary>
    /// Simulates loading items from the data source.
    /// </summary>
    /// <returns>A list of Item objects.</returns>
    public List<Item> LoadItems()
    {
        Console.WriteLine($"Simulating loading items from {_dataSourcePath}...");
        var items = new List<Item>
        {
            new Item(1, "Gadget Alpha", 150.75),
            new Item(2, "Widget Beta", 85.0),
            new Item(3, "Thingamajig Gamma", 210.5),
            new Item(4, "Doohickey Delta", 55.2)
        };

        Console.WriteLine($"Loaded {items.Count} items.");
        return items;
    }

    /// <summary>
    /// Simulates saving processed items back to the data source.
    /// </summary>
    /// <param name="items">A list of Item objects to save.</param>
    /// <returns>True if saving was simulated successfully.</returns>
    public bool SaveItems(List<Item> items)
    {
        Console.WriteLine($"Simulating saving {items.Count} items to {_dataSourcePath}...");
        foreach (var item in items)
        {
            Console.WriteLine($"Saving item: {item}");
        }
        Console.WriteLine("Finished simulating save operation.");
        return true;
    }
}